File: Python/number_factory/number_factory.py
import sys


class NumberFactory:
    def add(self, int_a, int_b):
        self._validate_int(int_a)
        self._validate_int(int_b)
        return int_a + int_b

    def subtract(self, int_a, int_b):
        self._validate_int(int_a)
        self._validate_int(int_b)
        return int_a - int_b

    def multiply(self, int_a, int_b):
        self._validate_int(int_a)
        self._validate_int(int_b)
        return int_a * int_b

    def divide(self, int_a, int_b):
        return float(int_a) / int_b

    def _validate_int(self, arg):
        if not isinstance(arg, int):
            raise ArgumentException("Argument: {} is not an int".format(int))


class ArgumentException(Exception):
    pass

File: Go/src/number_factory/number_factory.go
package number_factory


type NumberFactory struct {
    A, B int
    C []int
}

func (n NumberFactory) Add() int {
    return n.A + n.B
}

func (n NumberFactory) Subtract() int {
    return n.A - n.B
}

func (n NumberFactory) Multiply() int {
    return n.A * n.B
}

func (n NumberFactory) Divide() float64 {
    return float64(n.A) / float64(n.B)
}

func (n NumberFactory) Square() int {
    return n.A * n.A
}

func (n NumberFactory) Sum() int {
    sum := 0
    for i := 0; i < len(n.C); i++ {
        sum += n.C[i]
    }
    return sum
}
